joomla/joomla-cms
# Ability to add other attributes to the script tag

### Description

This PR adds the ability to add more attributes to the script tag using `JDocument` `addScript` or `addScriptVersion` methods.
### How to test

Add a external js script like, some examples:

```
JFactory::getDocument()->addScript('/path/to/external/jsfile.js', '', false, false, array('id' => 'scriptid'));

JFactory::getDocument()->addScript('/path/to/external/jsfile.js', '', false, false, array('id' => 'scriptid', 'data-test' => 1));

JFactory::getDocument()->addScriptVersion('/path/to/external/jsfile.js', 3, '', false, false, array('id' => 'scriptid'));

JFactory::getDocument()->addScript('/path/to/external/jsfile.js', '', false, false, array('data-attrib' => 5));

JFactory::getDocument()->addScript('/path/to/external/jsfile.js', 'text/javascript', true, false, array('data-attrib' => array('item1' => 3, 'item2' => 'value2')));

JHtml::_('script', 'system/jsfile.js', false, true, false, true, true, array('data-attrib' => array('item1' => 3, 'item2' => 'value2')));

JHtml::_('script', 'system/jsfile.js', false, true, false, true, true, array('defer' => 'defer', 'data-attrib' => array('item1' => 3, 'item2' => 'value2')));
```

Load the page and check the html code. The script tag should be rendered with the attribute(s) added.
### B/C

None that i know of.
